src/main/webapp/index.jsp

```diff
@@ -158,21 +117,55 @@
           li.textContent = place.name;
           placesList.appendChild(li);
           bounds.extend(place.geometry.location);
-          placesArray.push(place.name);
+          placesArray.push(place.place_id);
         }
         map.fitBounds(bounds);
-        var numResult = Math.floor(Math.random() * 20);
+        var numResult = Math.floor(Math.random() * (placesArray.length));
         var restaurantChoice = placesArray[numResult];
         alert(restaurantChoice);
-      }
+      }//createMarkers
+
+      function geocodeAddress(geocoder, resultsMap) {
+        const address = document.getElementById("address").value;
+        geocoder.geocode(
+          {
+            address: address
+          },
+          (results, status) => {
+            if (status === "OK") {
+              resultsMap.setCenter(results[0].geometry.location);
+              new google.maps.Marker({
+                map: resultsMap,
+                position: results[0].geometry.location
+              });
+            } else {
+              alert(
+                "Geocode was not successful for the following reason: " + status
+              );
+            }
+          }
+        );
+      }//geocodeAddress
+
     </script>
   </head>
   <body>
+    <div id="floating-panel">
+      <input id="address" type="textbox" value="Sydney, NSW" />
+      <input id="submit" type="button" value="Geocode" />
+    </div>
     <div id="map"></div>
+
+    <script
+      src="https://maps.googleapis.com/maps/api/js?key=AIzaSyBnHjJrXSDt16Vgg76YOzIBK1KTid0cfeI&callback=initMap&libraries=places&v=weekly"
+      defer
+    ></script>
```

Comment:
should the script tag be after the scripts? was it moved under the scripts to ensure that the functions run before the map is loaded?